### ISSUE TITLE
CPU hang while analyzing PyQt5

### ISSUE THREAD

**Issue by jakebailey** [Microsoft/python-language-server#875]
```python
import pandas as pd 
import datetime 
import pandas_datareader.data as web

start = datetime.datetime(2010, 1,1)
end = datetime.datetime.now()
df = web.DataReader("MSFT", "yahoo", start, end)



df.set_index("Date", inplace = True)

import matplotlib.pyplot as plt
from matplotlib import style

style.use('fivethirtyeight')


df['High'].plot()

plt.legend()
plt.show()
```

With:

[requirements.txt](https://github.com/Microsoft/python-language-server/files/3040754/requirements.txt)

**Comment by mhzed** [comment]
Similar issue here, it stays at 100% forever "Analyzing in background, xxx items left".   This started happening after updating to version "2019.3.6352".  Seems though it picked up the virtual env and got stuck analyzing the site-packages.

**Comment by ahuigo** [comment]
Same 100% cpu issue with my small project: 
1. It has been analyzing forever when I open vscode from code bin: \
`/usr/local/bin/code -> /Applications/Visual Studio Code.app/Contents/Resources/app/bin/code`
![image](https://user-images.githubusercontent.com/1781999/55526482-ddbad380-56c7-11e9-999f-0374d1594e32.png)
2. It will remind me to enable pylint every time even through I have enabled pylint.
3. Here is screen shot from `Development Tools`:
![image](https://user-images.githubusercontent.com/1781999/55526730-b0baf080-56c8-11e9-9fe0-a5c575508f10.png)
![image](https://user-images.githubusercontent.com/1781999/55527098-296e7c80-56ca-11e9-8fd8-e54a425f7c2a.png)


## Environment data
With Python 3.7.2 installed:
```
vscode 1.32.3 
Mac Mojave 10.14
pylint 2.3.1
astroid 2.2.5
Python 3.7.2 (default, Feb 12 2019, 08:15:36)

➜ pydemo$ code --list-extensions --show-versions
batisteo.vscode-django@0.19.0
donjayamanne.jupyter@1.1.5
donjayamanne.python-extension-pack@1.4.0
magicstack.MagicPython@1.1.0
ms-python.python@2019.3.6215
ms-vscode.Go@0.9.2
ms-vsliveshare.vsliveshare@0.3.1403
msjsdiag.debugger-for-chrome@4.11.3
mushan.vscode-paste-image@1.0.4
Shan.code-settings-sync@3.2.7
tht13.python@0.2.3
VisualStudioExptTeam.vscodeintellicode@1.1.5
vscodevim.vim@1.3.0
wholroyd.jinja@0.0.8
yzane.markdown-pdf@1.2.0
yzhang.markdown-all-in-one@2.2.0
````
Related issue:
https://github.com/Microsoft/vscode-python/issues/5074

**Comment by MikhailArkhipov** [comment]
@ahuigo, @mhzed - what is the LS version (displayed at startup in the output). 0.2.43 is latest.

**Comment by ahuigo** [comment]
@MikhailArkhipov languageServer.0.2.41
```
[Extension Host] Python Extension: Get language server folder name, Class name = g, , Return Value: "languageServer.0.2.41"
```
Any more should I provide?

**Comment by mhzed** [comment]
Mine was 0.2.41.0
Just updated to 0.2.43.0 and now the problem appears to be gone.

**Comment by ahuigo** [comment]
@mhzed How to update it to 0.2.43.0 ?

**Comment by MikhailArkhipov** [comment]
Delete `languageServer.*` folder under extension (use `Open Extensions Folder` command)

**Comment by ahuigo** [comment]
```mv /Users/ahui/.vscode/extensions/ms-python.python-2019.3.6352/languageServer.0.2.41 ~/backup
```
Remove 0.2.41 and updated to 0.2.43, the problem disappears now.

**Comment by eyadsibai** [comment]
it fixes the problem temporarily, it starts again after opening/closing couple of projects

**Comment by ionlz** [comment]
I still have this problem that high CPU usage.And if i enable pylint, the vscode will take up a lot of memory.
I have removed 0.2.41.

**Comment by MikhailArkhipov** [comment]
It depends on what your code imports. For example, a single tensorflow import effectively means 3000 files to be analyzed. Which takes CPU cycles. Although we are working on additional optimizations and caching, in some cases there is little way around the fact that some libraries are very big with circular imports.

**Comment by ahuigo** [comment]
If I import nodejs and golang directory , will it analyze node_modules and go files?

**Comment by MikhailArkhipov** [comment]
No, of course not. Only py, pyd and other related Python files unless there are stubs, in which case the stubs are used rather than the library source.

**Comment by ahuigo** [comment]
```
# config.py
name = "Jack"
```

Could not go to  variable's definition: 


```
# main.py
# Could not go to  `name` definition: 
from config import name
```

**Comment by jakebailey** [comment]
This thread is about CPU usage, if you believe you have a different issue we'd appreciate it if you could make another thread so we can handle it separately. Thanks!

**Comment by jakebailey** [comment]
Root cause appears to be #907.

**Comment by jakebailey** [comment]
The hang that I was investigating in this issue appears to be fixed now that #942 is merged; we still analyze all of PyQt5 which takes a few seconds, but it's no longer taking minutes.

If there are other examples where the analysis appears to not complete then they should go into another issue (as I'd like to close this specific item as done).
